### PR TITLE
Bump uniffi-rs dependency to 0.23.0

### LINKED ISSUE #301: Long-term solution exploration for generating correct Python wheels
As explained [here](https://github.com/mozilla/uniffi-rs/issues/1307), the Python scaffolding code does not currently find the native binaries on its own. We have a workaround, but don't think it's an optimal solution for the long run.

After some discussion with the team at Mozilla, I'm opening this issue to keep track of different approaches that could solve our issue.

1. The solution implemented by [Glean in their setup.py file](https://github.com/mozilla/glean/blob/main/glean-core/python/setup.py)
2. Leveraging [maturin](https://github.com/pyo3/maturin/).

### PR DESCRIPTION
**Pull request by thunderbiscuit**
## Description
This PR updates the uniffi-rs library dependency to the 0.23.0 version.

## Notes to the reviewers
There are significant changes in this release which will impact our codebase:
1. The only dependency is on uniffi (no more `uniffi_bindgen` nor `uniffi_build`).
2. It might allow us to deprecate the use of the bdk-ffi-bindgen tool. I'm not sure about this just yet, I have performed the migration for Android, JVM, and Python, but still need to do Swift.
3. In the process of migrating the Python build workflow, I believe I have found the solution to finally get rid of rust-setuptools and remove the need for the bespoke inline change in the glue code to allow for loading the libraries. If the new workflow passes the CI and is tested locally and working on all 3 platforms by contributors, we will be able to close #301 (see mozilla/uniffi-rs#1307 for more info).

### Changelog notice
```md
Update uniffi-rs to version 0.23.0 [#299]

[#299]: https://github.com/bitcoindevkit/bdk-ffi/pull/299
```

#### All Submissions:
* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
After a ferocious battle with the CI, this is ready for review @notmandatory. Lots of small changes but a really good step forward.

**Comment by notmandatory** [comment]
I've finished manual testing with BdkSwiftSample and after making required API changes it all works fine, see futurepaul/BdkSwiftSample#11.

https://github.com/futurepaul/BdkSwiftSample/pull/11

File: .github/workflows/publish-python.yaml [hunk 1]
```diff
@@ -98,45 +104,44 @@ jobs:
...
-      - run: pip install --user -r requirements.txt
-      - run: bash generate.sh
-        shell: bash
-      - run: pip install --user wheel
-      - name: build wheel
-        run: python setup.py -v bdist_wheel
+
+      - name: "Generate bdk.py"
+        run: |
+          python --version
```
Comment (notmandatory) [inline .github/workflows/publish-python.yaml:117]: Just a nit.. but you don't need this line, you probably only had it in for debugging.

File: .github/workflows/test-python.yaml [hunk 2]
```diff
@@ -71,30 +78,35 @@ jobs:
...
-      - run: python3 --version
-      - run: rustup target add aarch64-apple-darwin
-      - run: pip3 install --user -r requirements.txt
-      - run: pip3 install --user wheel
-      - run: bash generate.sh
-      - name: build wheel
+
+      - name: "Generate bdk.py"
+        run: | 
+          python3 --version
```
Comment (notmandatory) [inline .github/workflows/test-python.yaml:91]: nit, another line that can be removed

**Comment by thunderbiscuit** [comment]
Awesome! Glad we can do a clean migration and not have to keep some languages on the 0.21 and some upgraded to the 0.23 build workflow.

Mind giving an ACK for the PR? I'll merge and we can start playing with macros!

**Review by notmandatory — APPROVED** [review]
ACK c1243f9e1c9ff6b5661ac54d799fc2f0d96232a0

I reviewed the changes and they make sense to me, though I don't know the Python side. I only manually tested the `build-local-swift.sh` script.  I'm counting on CI for the rest of the tests.

**Pull request merged**